File: src/main/java/com/falsepattern/zigbrains/zon/parser/ZonFile.java
package com.falsepattern.zigbrains.zon.parser;

import com.falsepattern.zigbrains.zon.ZonFileType;
import com.falsepattern.zigbrains.zon.ZonLanguage;
import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

public class ZonFile extends PsiFileBase {
    public ZonFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, ZonLanguage.INSTANCE);
    }

    @Override
    public @NotNull FileType getFileType() {
        return ZonFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Zon File";
    }
}

File: modules/core/src/main/kotlin/com/falsepattern/zigbrains/zon/psi/ZonFile.kt
package com.falsepattern.zigbrains.zon.psi

import com.falsepattern.zigbrains.zon.ZonFileType
import com.falsepattern.zigbrains.zon.ZonLanguage
import com.intellij.extapi.psi.PsiFileBase
import com.intellij.psi.FileViewProvider

class ZonFile(viewProvider: FileViewProvider): PsiFileBase(viewProvider, ZonLanguage) {
    override fun getFileType() = ZonFileType

    override fun toString() = "Zon File"
}